src/pages/index.js

```diff
@@ -80,64 +74,185 @@ const addCardPopup = new PopupWithForm(
+});
+
+// 
+// Functions
+//  
+
+function createCard(cardData) {
+	const card = new Card(
+		cardData,
+		'#card-template',
+		handleImageClick,
+		handleTrashClick,
+		handleLikeClick,
+	);
+	cardSection.addItem(card.generateCard());
 }
 
-function handleImageClick(name, link) {
-	popupImageModal.open(name, link);
+// ADD CARD AND EDIT PROFILE
+
+function handleAddCardFormSubmit(data) {
+	addCardPopup.submitButtonState(true, "Saving...");
+	api
+		.addCard(data)
+		.then((data) => {
+			createCard(data);
+		})
+		.then(() => {
+			addCardPopup.close();
+		})
+		.catch((err) => {
+			console.log(err);
+		})
+		.finally(() => {
+			addCardPopup.submitButtonState(false);
+		});
 }
 
-function handleProfileEditSubmit(value) {
-	userInfo.setUserInfo(value);
-	editProfilePopup.close();
+function handleProfileEditSubmit(data) {
+	editProfilePopup.submitButtonState(true, "Saving...");
+	api
+	.updateEditProfile(data)
+		.then((data) => {
+			userInfo.setUserInfo(data);
+			editProfilePopup.close();
+		})
+		.catch((err) => {
+			console.log(err);
+		})
+		.finally(() => {
+			editProfilePopup.submitButtonState(false);
+		});
+}
+
+//  
+//  Avatar
+//  
+
+const avatarPopup = new PopupWithForm(
+	'#avatar-modal',
+	handleAvatarFormSubmit
+);
+avatarPopup.setEventListeners();
+
+function handleAvatarFormSubmit(data) {
+	avatarPopup.submitButtonState(true, "Saving...");
+	api
+	.updateAvatar(data)
+	  .then((data) => {
+		// userInfo.setUserInfo(data);
+		userInfo.setAvatar(data.avatar);
+		avatarPopup.close();
+	  })
+	  .catch((err) => {
+		console.log(err);
+	  })
+	  .finally(() => {
+		avatarPopup.submitButtonState(false);
+	  });
+}
+
+const avatarButton = document.querySelector(".avatar__edit_button");
+avatarButton.addEventListener("click", () => {
+  updateAvatarValidator.toggleButtonState();
+  avatarPopup.open();
+});
+
+//  
+//  Like Card
+//
+
+function handleLikeClick(card) {
+  if (card.isLiked) {
+	api
+	  .unlikeCard(card.id)
+	  .then(() => {
+		card.updateIsLiked(false)
+	  });
+	} else {
+	api
+	  .likeCard(card.id)
+	  .then(() => {
+		card.updateIsLiked(true)
+	  })
+	  .catch((err) => {
+		console.log(err);
+	  });
+  }
 }
-function handleAddCardFormSubmit(value) {
-	const name = value.title
-	const link = value.link
-	renderCard({ name, link }, cardListEl);
-	addCardFormElement.reset();
-	addCardPopup.close();
+
+//  
+// Confirm Delete Popup
+//
+const confirmDeletePopup = new PopupConfirmation({
+	popupSelector: "#delete-card-modal"
+	});
+confirmDeletePopup.setEventListeners();
+
+  function handleTrashClick(card) {
+	confirmDeletePopup.open();
+	confirmDeletePopup.setSubmitCallback(() => {
+		api
+	.deleteCard(card.id)
+	.then(() => {
+		card.handleRemoveCard();
+		confirmDeletePopup.close();
+	})
+	.catch((err) => {
+		console.log(err);
+	});
+
+  });
 }
+  
+//
+// Overlay Close
+//  
 
+  function handleOverlayClose(e) {
+	if (e.target.classList.contains('modal_opened')) {
+		popupImageModal.close();
+	}
```

Comment:
Please, delete overlay handling from `indexj.s`. Everything is handled by `Popup` from now on

Reply:
I deleted the function handleOverlayClose and the corresponding event listeners in index.js.